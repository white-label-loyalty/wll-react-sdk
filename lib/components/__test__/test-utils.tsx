import { render } from '@testing-library/react';
import React from 'react';
import { CoreApiUrl, WllSdkProvider } from '../../context/WllSdkContext';
import { defaultTheme } from '../../utils/styling';

const defaultConfig = {
  apiKey: 'test-api-key',
  userToken: 'test-user-token',
  coreApiUrl: CoreApiUrl.STAGING_EU,
};

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <WllSdkProvider
      config={defaultConfig}
      theme={defaultTheme}
      navigationConfig={{}}
    >
      {children}
    </WllSdkProvider>
  );
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
