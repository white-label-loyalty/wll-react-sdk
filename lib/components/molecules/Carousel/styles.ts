import { StyleSheet } from 'react-native';
import { MAX_WIDTH } from '../../../constants';
import { useWllSdk } from '../../../context/WllSdkContext';
import { useResponsive } from '../../../hooks/useResponsive';
import { useResponsiveValue } from '../../../utils/responsiveHelper';

export const useCarouselStyles = (buttonSize = 42, slideWidth: number) => {
  const { isDesktop, isTablet } = useResponsive();
  const { theme } = useWllSdk();

  return StyleSheet.create({
    container: {
      flex: 1,
      maxWidth: MAX_WIDTH,
    },
    sectionTitle: {
      fontSize: 31,
      marginBottom: 10,
    },
    sectionDescription: {
      marginBottom: 21,
    },
    title: {
      marginBottom: 10,
    },
    description: {
      marginBottom: 20,
    },
    carouselContainer: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
    },
    carouselContent: {
      overflow: 'hidden',
      width: slideWidth,
    },
    navButton: {
      position: 'absolute',
      top: '50%',
      transform: [{ translateY: -buttonSize / 2 }],
      width: buttonSize,
      height: buttonSize,
      borderRadius: buttonSize / 2,
      backgroundColor: theme.background,
      justifyContent: 'center',
      alignItems: 'center',
      zIndex: 1,
    },
    navButtonLeft: {
      left: -buttonSize / 2,
    },
    navButtonRight: {
      right: -buttonSize / 2,
      backgroundColor: theme.surface,
    },
    slideContainer: {
      width: slideWidth,
    },
    indicators: {
      flexDirection: 'row',
      justifyContent: 'center',
      marginTop: useResponsiveValue(24, 12, isDesktop, isTablet),
    },
    indicator: {
      width: 8,
      height: 8,
      borderRadius: 8,
      marginHorizontal: 4,
      backgroundColor: theme.derivedBackground,
    },
    activeIndicator: {
      backgroundColor: theme.primary,
    },
  });
};
