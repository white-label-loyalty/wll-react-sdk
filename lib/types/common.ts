import { ImageProps, ViewStyle } from 'react-native';

export type ImagePropsNoSource = Omit<ImageProps, 'source'>;

export type FlexDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';

export type Justify =
  | 'start'
  | 'end'
  | 'center'
  | 'between'
  | 'around'
  | 'evenly';

export type Align = 'start' | 'end' | 'center' | 'stretch' | 'baseline';

export type LayoutProps = {
  children: React.ReactNode;
  justify?: Justify;
  align?: Align;
  direction?: FlexDirection;
  style?: ViewStyle;
};

export const justifyMap: Record<Justify, ViewStyle['justifyContent']> = {
  start: 'flex-start',
  end: 'flex-end',
  center: 'center',
  between: 'space-between',
  around: 'space-around',
  evenly: 'space-evenly',
} as const;

export const alignMap: Record<Align, ViewStyle['alignItems']> = {
  start: 'flex-start',
  end: 'flex-end',
  center: 'center',
  stretch: 'stretch',
  baseline: 'baseline',
} as const;
