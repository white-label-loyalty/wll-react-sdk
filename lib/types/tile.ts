export enum CtaAction {
  sameWindow = 'SAME_WINDOW',
  newWindow = 'NEW_WINDOW',
}

export enum BadgeTileType {
  latestEarned = 'LATEST_EARNED',
  specificBadge = 'SPECIFIC_BADGE',
}

export enum TierTileType {
  currentTier = 'CURRENT_TIER',
  nextTier = 'NEXT_TIER',
  specificTier = 'SPECIFIC_TIER',
}

export enum TileType {
  Banner = 'BANNER',
  Points = 'POINTS',
  Content = 'CONTENT',
  Reward = 'REWARD',
  Badge = 'BADGE',
  RewardCategory = 'REWARD_CATEGORY',
  Tier = 'TIER',
}

export enum TileHeight {
  Half = 'HALF',
  Full = 'FULL',
}
type Badge = {
  id: string | null;
  name: string | null;
  description: string | null;
  artworkUrl: string | null;
  createdAt: string | null;
  updatedAt: string | null;
};
export class RewardCategory {
  name?: string | null;
  priority?: number;
  type?: string | null;
  id?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  description?: string | null;
  metadata?: string | null;
  pictureUrl?: string | null;
  rewards?: Reward[];
  parent?: string | null;
}

export class Reward {
  id?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  name?: string | null;
  pictureUrl?: string | null;
  description?: string | null;
  value?: number | null;
  price?: number;
  priority?: number;
  discounts?: any[] | null;
  redemptionChannels?: string[] | null;
  venues?: any[] | null;
  purchasableForAudiences?: string[] | null;
  category?: string | null;
  terms?: string | null;
  availability?: Availability;
  purchasable?: boolean;
  tier?: string | null;
  summary?: string | null;
  redemptionMessage?: string | null;
  visibilityCriteria?: string | null;
  logoUrl?: string | null;
  type?: string | null;
  codeType?: string | null;
  code?: string | null;
  purchaseExpiration?: string | null;
  hideCode?: boolean;
  notificationConfig?: string | null;
}
type Availability = {
  start: string | null;
  end: string | null;
};
type Effectivity = {
  start: string | null;
  end: string | null;
};

export type TierType = {
  id: string | null;
  name: string | null;
  description: string | null;
  artworkUrl: string | null;
  priority: number;
  pointsRequirement: number;
  calculation: string | null;
  accumulationPeriod: string | null;
  effectivity: Effectivity;
  createdAt: string | null;
  updatedAt: string | null;
  earnedPoints: number;
  attained: boolean;
};

export class BannerTileConfig {
  imageUrl?: string | null;
  title?: string | null;
  description?: string | null;
  ctaText?: string | null;
  ctaLink?: string | null;
  ctaAction?: CtaAction;
}

export class PointsTileConfig {
  title?: string | null;
  multiplier?: number;
  prefix?: string | null;
  suffix?: string | null;
  imageUrl?: string | null;
  points?: number;
}

export class ContentTileConfig {
  title?: string | null;
  subtitle?: string | null;
  imageUrl?: string | null;
}

export class RewardTileConfig {
  reward?: Reward;
  showPrice?: boolean;
}

export class BadgeTileConfig {
  badgeTileType?: BadgeTileType;
  badgeId?: string | null;
  badge?: Badge;
}

export class RewardCategoryTileConfig {
  categoryId?: string | null;
  allowDecorationOverlay?: boolean;
  rewardCategory?: RewardCategory;
}

export class TierTileConfig {
  tierId?: string | null;
  tier?: TierType;
  type?: TierTileType;
}

export type Tile = {
  id: string | null;
  type: TileType;
  active: boolean;
  createdAt: string | null;
  updatedAt: string | null;
  visibilityCriteria: object | null;
  tileHeight: TileHeight;
  configuration: TileConfig;
};

const getConfigForTileType = (tileType: TileType) => {
  switch (tileType) {
    case TileType.Banner:
      return BannerTileConfig;
    case TileType.Points:
      return PointsTileConfig;
    case TileType.Content:
      return ContentTileConfig;
    case TileType.Reward:
      return RewardTileConfig;
    case TileType.Badge:
      return BadgeTileConfig;
    case TileType.RewardCategory:
      return RewardCategoryTileConfig;
    case TileType.Tier:
      return TierTileConfig;
    default:
      throw new Error(`Tile type ${tileType} is not supported`);
  }
};

export type TileConfig = InstanceType<
  ReturnType<typeof getConfigForTileType>
>;
