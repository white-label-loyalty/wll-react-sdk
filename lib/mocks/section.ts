import { SectionType, TSection } from '../types/section';
import { Tile, TileType } from '../types/tile';
import { createTileMock } from './tiles';

export type SectionMockConfig = Partial<TSection> & {
  tiles?: Tile[];
};

const defaultConfig: TSection = {
  details: [],
  defaultLocale: 'en',
  name: 'Elite Member Benefits',
  type: SectionType.Grid,
  active: true,
  id: 'a35399a7-580a-4493-afff-24ef4facf3f4',
  createdAt: '2024-11-19T15:50:37.701Z',
  updatedAt: '2024-12-06T11:18:16.842Z',
  priority: 2,
  tiles: [createTileMock(TileType.Content)],
  title: 'Elite Member Benefits',
  locale: 'en',
  pointsSuffix: 'miles',
};

export const createSectionMock = (config?: SectionMockConfig): TSection => {
  const { tiles, ...sectionConfig } = config || {};
  
  // If tiles are provided in config, use those. Otherwise use default tiles
  const mergedTiles = tiles || defaultConfig.tiles;

  return {
    ...defaultConfig,
    ...sectionConfig,
    tiles: mergedTiles,
  };
};

// Helper function to create a section with specific tile types
export const createSectionWithTiles = (tiles: Tile[]): TSection => {
  return createSectionMock({ tiles });
};
