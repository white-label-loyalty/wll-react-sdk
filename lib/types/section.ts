import { Tile } from './tile';

export enum SectionType {
  Grid = 'GRID',
  Banner = 'BANNER',
}

export type TSection = {
  id: string;
  type: SectionType;
  createdAt: string;
  updatedAt: string;
  title?: string;
  description?: string;
  tiles: Tile[];
  pointsMultiplier?: number;
  pointsPrefix?: string | null;
  pointsSuffix?: string | null;
  priority: number;
  active: boolean;
};
