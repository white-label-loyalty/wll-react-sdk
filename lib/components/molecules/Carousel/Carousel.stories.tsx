import { Meta, StoryFn } from '@storybook/react';
import React from 'react';
import { Section, SectionType } from '../../../types/section';
import {
  BannerTileConfig,
  Tile,
  TileHeight,
  TileType,
  UrlTarget,
} from '../../../types/tile';
import Carousel from './index';

export default {
  title: 'components/molecules/Carousel',
  component: Carousel,
  argTypes: {
    section: { control: 'object' },
  },
} as Meta;

const Template: StoryFn<typeof Carousel> = (args) => <Carousel {...args} />;

const createMockTile = (config: Partial<BannerTileConfig>): Tile => ({
  id: Math.random().toString(36).substr(2, 9),
  type: TileType.Banner,
  createdAt: 'test-date',
  updatedAt: 'test-date',
  active: true,
  tileHeight: TileHeight.Full,
  configuration: {
    imageUrl: config.imageUrl,
    title: config.title,
    description: config.description,
    buttonText: config.buttonText,
    url: config.url,
    urlTarget: config.urlTarget,
  },
});

const createMockSection = (
  title: string,
  description: string,
  tiles: Tile[]
): Section => ({
  id: Math.random().toString(36).substr(2, 9),
  title,
  type: SectionType.Banner,
  active: true,
  priority: 0,
  description,
  tiles,
  createdAt: 'test-date',
  updatedAt: 'test-date',
});

export const Default = Template.bind({});
Default.args = {
  section: createMockSection(
    'Featured Offers',
    'Check out our latest promotions and deals!',
    [
      createMockTile({
        title: 'Refer a friend',
        imageUrl:
          'https://cdn.pixabay.com/photo/2014/06/03/19/38/board-361516_1280.jpg',
        description:
          'Click here to share your unique referral link with friends!',
        buttonText: 'REFER NOW',
        urlTarget: UrlTarget.sameWindow,
      }),
      createMockTile({
        title: 'Summer Sale',
        imageUrl:
          'https://cdn.pixabay.com/photo/2014/06/03/19/38/board-361516_1280.jpg',
        description:
          'Enjoy up to 50% off on selected items. Limited time offer!',
        buttonText: 'SHOP NOW',
        urlTarget: UrlTarget.newWindow,
        url: 'https://www.example.com',
      }),
      createMockTile({
        title: 'New Collection',
        imageUrl:
          'https://cdn.pixabay.com/photo/2014/06/03/19/38/board-361516_1280.jpg',
        description: 'Check out our latest arrivals for this season.',
        buttonText: 'EXPLORE',
        urlTarget: UrlTarget.sameWindow,
        url: 'https://www.example.com',
      }),
    ]
  ),
};

export const SingleItem = Template.bind({});
SingleItem.args = {
  section: createMockSection(
    'Special Promotion',
    "Don't miss out on this exclusive offer!",
    [
      createMockTile({
        title: 'Special Offer',
        description: 'Get a free gift with any purchase over $50.',
        buttonText: 'CLAIM NOW',
        imageUrl:
          'https://cdn.pixabay.com/photo/2014/06/03/19/38/board-361516_1280.jpg',
        urlTarget: UrlTarget.sameWindow,
        url: 'https://www.example.com',
      }),
    ]
  ),
};

export const ManyItems = Template.bind({});
ManyItems.args = {
  section: createMockSection(
    'All Promotions',
    'Browse through our extensive list of current promotions.',
    Array(10)
      .fill(null)
      .map((_, index) =>
        createMockTile({
          title: `Offer ${index + 1}`,
          imageUrl:
            'https://cdn.pixabay.com/photo/2014/06/03/19/38/board-361516_1280.jpg',
          description: `This is the description for offer number ${index + 1}.`,
          buttonText: `ACTION ${index + 1}`,
          urlTarget:
            index % 2 === 0 ? UrlTarget.sameWindow : UrlTarget.newWindow,
        })
      )
  ),
};
