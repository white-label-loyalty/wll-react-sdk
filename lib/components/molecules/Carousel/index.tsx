import React, { useCallback, useRef, useState } from 'react';
import {
  Animated,
  NativeScrollEvent,
  NativeSyntheticEvent,
  ScrollView,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native';
import { BUTTON_SIZE, SLIDE_WIDTH } from '../../../constants';
import { useResponsive } from '../../../context/ResponsiveContext';
import { useWllSdk } from '../../../context/WllSdkContext';
import { TSection } from '../../../types/section';
import { Tile, TileType } from '../../../types/tile';
import { getResponsiveValue } from '../../../utils/responsiveHelper';
import { sortByPriority } from '../../../utils/transforms';
import { Icon } from '../../atoms';
import { BannerTile } from '../../organisms';
import SectionHeader from '../SectionHeader';
import { useCarouselStyles } from './styles';

type CarouselProps = {
  section: TSection;
};

const Carousel: React.FC<CarouselProps> = ({ section }) => {
  const styles = useCarouselStyles(BUTTON_SIZE, SLIDE_WIDTH);
  const animatedIndex = useRef(new Animated.Value(0)).current;
  const { theme } = useWllSdk();
  const { isDesktop, isTablet } = useResponsive();
  const scrollViewRef = useRef<ScrollView>(null);
  const [currentIndex, setCurrentIndex] = useState(0);

  const bannerTiles = section.tiles.filter(
    (tile: Tile) => tile.type === TileType.Banner
  );
  const sortedTiles = sortByPriority(bannerTiles);

  const handleScroll = useCallback(
    (event: NativeSyntheticEvent<NativeScrollEvent>) => {
      const contentOffsetX = event.nativeEvent.contentOffset.x;
      const newIndex = contentOffsetX / SLIDE_WIDTH;
      animatedIndex.setValue(newIndex);
    },
    [SLIDE_WIDTH, animatedIndex]
  );

  const handleScrollEnd = useCallback(
    (event: NativeSyntheticEvent<NativeScrollEvent>) => {
      const contentOffsetX = event.nativeEvent.contentOffset.x;
      const newIndex = Math.round(contentOffsetX / SLIDE_WIDTH);
      setCurrentIndex(newIndex);
    },
    [SLIDE_WIDTH]
  );

  const handlePrev = () => {
    const newIndex = Math.max(0, currentIndex - 1);
    setCurrentIndex(newIndex);
    scrollViewRef.current?.scrollTo({
      x: newIndex * SLIDE_WIDTH,
      animated: true,
    });
  };

  const handleNext = () => {
    const newIndex = Math.min(sortedTiles.length - 1, currentIndex + 1);
    setCurrentIndex(newIndex);
    scrollViewRef.current?.scrollTo({
      x: newIndex * SLIDE_WIDTH,
      animated: true,
    });
  };

  const displayControls = sortedTiles.length > 1;

  const dynamicStyles = StyleSheet.create({
    indicators: {
      flexDirection: 'row',
      justifyContent: 'center',
      marginTop: getResponsiveValue(24, 12, isDesktop, isTablet),
    },
  });

  return (
    <>
      <SectionHeader title={section.title} description={section.description} />
      <View style={styles.container}>
        <View style={styles.carouselContainer}>
          {displayControls && (
            <TouchableOpacity
              style={[
                styles.navButton,
                styles.navButtonLeft,
                { backgroundColor: theme.background },
              ]}
              onPress={handlePrev}
            >
              <Icon name="ArrowLeft" size={20} color={theme.primary} />
            </TouchableOpacity>
          )}
          <ScrollView
            ref={scrollViewRef}
            horizontal
            pagingEnabled
            showsHorizontalScrollIndicator={false}
            onScroll={handleScroll}
            onMomentumScrollEnd={handleScrollEnd}
            scrollEventThrottle={16}
            style={[styles.carouselContent, { width: SLIDE_WIDTH }]}
            contentContainerStyle={{
              width: SLIDE_WIDTH * sortedTiles.length,
            }}
            decelerationRate="fast"
            snapToInterval={SLIDE_WIDTH}
            snapToAlignment="start"
          >
            {sortedTiles.map((tile: Tile, index: number) => (
              <View
                key={index}
                style={[
                  {
                    width: SLIDE_WIDTH,
                  },
                ]}
              >
                <BannerTile tile={tile} />
              </View>
            ))}
          </ScrollView>
          {displayControls && (
            <TouchableOpacity
              style={[
                styles.navButton,
                styles.navButtonRight,
                { backgroundColor: theme.surface },
              ]}
              onPress={handleNext}
            >
              <Icon name="ArrowRight" size={20} color={theme.primary} />
            </TouchableOpacity>
          )}
        </View>
        {displayControls && (
          <View style={dynamicStyles.indicators}>
            {sortedTiles.map((_, index) => {
              const width = animatedIndex.interpolate({
                inputRange: [index - 1, index, index + 1],
                outputRange: [8, 30, 8],
                extrapolate: 'clamp',
              });
              return (
                <Animated.View
                  key={index}
                  style={[
                    styles.indicator,
                    { backgroundColor: theme.derivedBackground, width },
                    index === currentIndex && {
                      backgroundColor: theme.primary,
                    },
                  ]}
                />
              );
            })}
          </View>
        )}
      </View>
    </>
  );
};

export default Carousel;
