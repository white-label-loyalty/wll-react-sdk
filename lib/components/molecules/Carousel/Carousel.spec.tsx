import '@testing-library/jest-dom';
import React from 'react';
import { SectionType } from '../../../types/section';
import { TileHeight, TileType } from '../../../types/tile';
import { render, within } from '../../__test__/test-utils';
import Carousel from './index';

const mockSection = {
  id: 'test-section',
  name: 'Banner Section',
  type: SectionType.Banner,
  active: true,
  createdAt: '2025-01-27T18:31:01Z',
  updatedAt: '2025-01-27T18:31:01Z',
  visibilityCriteria: null,
  priority: 5,
  title: 'Featured Banners',
  description: 'Check out our latest promotions and updates!',
  defaultLocale: 'en',
  locale: 'en',
  tiles: [
    {
      id: '1',
      type: TileType.Banner,
      active: true,
      tileHeight: TileHeight.Full,
      priority: 2,
      configuration: {
        title: 'First Banner',
        description: 'First Banner Description',
      },
      createdAt: '2025-01-27T18:31:01Z',
      updatedAt: '2025-01-27T18:31:01Z',
    },
    {
      id: '2',
      type: TileType.Badge,
      active: true,
      tileHeight: TileHeight.Full,
      priority: 1,
      configuration: {
        name: 'Badge Title',
        description: 'Badge Description',
      },
      createdAt: '2025-01-27T18:31:01Z',
      updatedAt: '2025-01-27T18:31:01Z',
    },
    {
      id: '3',
      type: TileType.Banner,
      active: true,
      tileHeight: TileHeight.Full,
      priority: 3,
      configuration: {
        title: 'Second Banner',
        description: 'Second Banner Description',
      },
      createdAt: '2025-01-27T18:31:01Z',
      updatedAt: '2025-01-27T18:31:01Z',
    },
  ],
};

describe('<Carousel />', () => {
  it('matches snapshot', () => {
    const { container } = render(<Carousel section={mockSection} />);
    expect(container).toMatchSnapshot();
  });

  it('renders section header', () => {
    const { getByText } = render(<Carousel section={mockSection} />);

    expect(getByText('Featured Banners')).toBeInTheDocument();
    expect(
      getByText('Check out our latest promotions and updates!')
    ).toBeInTheDocument();
  });

  it('filters out non-banner tiles', () => {
    const { queryByText } = render(<Carousel section={mockSection} />);

    expect(queryByText('First Banner')).toBeInTheDocument();
    expect(queryByText('Second Banner')).toBeInTheDocument();
    expect(queryByText('Badge Title')).not.toBeInTheDocument();
  });

  it('sorts banner tiles by priority', () => {
    const { getAllByTestId } = render(<Carousel section={mockSection} />);
    const bannerTiles = getAllByTestId('banner-tile');

    expect(bannerTiles).toHaveLength(2);
    expect(
      within(bannerTiles[0]).getByText('Second Banner')
    ).toBeInTheDocument();
    expect(
      within(bannerTiles[1]).getByText('First Banner')
    ).toBeInTheDocument();
  });

  it('shows next button when multiple banner tiles exist and carousel is at first slide', () => {
    const { getByRole } = render(<Carousel section={mockSection} />);

    expect(getByRole('button', { name: /next slide/i })).toBeInTheDocument();
  });

  it('hides navigation controls when only one banner tile exists', () => {
    const singleBannerSection = {
      ...mockSection,
      tiles: [mockSection.tiles[0]],
    };
    const { queryByRole } = render(<Carousel section={singleBannerSection} />);

    expect(
      queryByRole('button', { name: /previous slide/i })
    ).not.toBeInTheDocument();
    expect(
      queryByRole('button', { name: /next slide/i })
    ).not.toBeInTheDocument();
  });
});
