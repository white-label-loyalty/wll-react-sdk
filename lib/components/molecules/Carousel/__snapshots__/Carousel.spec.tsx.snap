// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> matches snapshot 1`] = `
<div>
  <div
    class="css-view-175oi2r r-marginBottom-1ifxtd0"
  >
    <div
      class="css-text-146c3p1 r-fontSize-1ui5ee8 r-marginBottom-5oul0u"
      dir="auto"
      style="font-weight: 700; color: rgb(0, 0, 0);"
    >
      Featured Banners
    </div>
    <div
      class="css-text-146c3p1 r-fontSize-1x35g6"
      dir="auto"
      style="color: rgba(0, 0, 0, 0.8);"
    >
      Check out our latest promotions and updates!
    </div>
  </div>
  <div
    class="css-view-175oi2r r-flex-13awgt0 r-width-13qz1uu"
  >
    <div
      class="css-view-175oi2r r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1777fci r-position-bnwqim r-width-13qz1uu"
    >
      <div
        class="css-view-175oi2r r-WebkitOverflowScrolling-150rngu r-flexDirection-18u37iz r-flexGrow-16y2uox r-flexShrink-1wbh5a2 r-overflowX-lltvgl r-overflowY-buy8e9 r-transform-agouwx r-scrollSnapType-mfh4gg r-overflow-1udh08x r-width-13qz1uu r-scrollbarWidth-2eszeu"
      >
        <div
          class="css-view-175oi2r r-flexDirection-18u37iz"
          style="width: 0px;"
        >
          <div
            class="css-view-175oi2r r-scrollSnapAlign-cpa5s6"
          >
            <div
              class="css-view-175oi2r"
              style="width: 0px;"
            >
              <button
                aria-disabled="true"
                aria-label="Second Banner"
                class="css-view-175oi2r r-pointerEvents-12vffkv r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1wtj0ep r-maxWidth-q90dru r-overflow-1udh08x r-width-13qz1uu"
                data-testid="banner-tile"
                disabled=""
                role="button"
                style="background-color: rgb(248, 247, 252); border-top-left-radius: 20px; border-top-right-radius: 20px; border-bottom-right-radius: 20px; border-bottom-left-radius: 20px; opacity: 1;"
                tabindex="-1"
                type="button"
              >
                <div
                  class="css-view-175oi2r"
                  style="flex: 1;"
                >
                  <div
                    aria-label="Second Banner"
                    class="css-view-175oi2r"
                  >
                    <div
                      class="css-text-146c3p1 r-fontSize-1ui5ee8 r-fontWeight-b88u0q r-marginBottom-6gpygo"
                      data-testid="banner-tile-title"
                      dir="auto"
                      style="color: rgb(0, 0, 0);"
                    >
                      Second Banner
                    </div>
                  </div>
                  <div
                    aria-label="Second Banner Description"
                    class="css-view-175oi2r"
                    data-testid="banner-tile-description"
                  >
                    <div
                      class="css-text-146c3p1 r-fontSize-1i10wst r-marginBottom-zd98yo"
                      dir="auto"
                      style="color: rgb(51, 51, 51);"
                    >
                      Second Banner Description
                    </div>
                  </div>
                </div>
              </button>
            </div>
          </div>
          <div
            class="css-view-175oi2r r-scrollSnapAlign-cpa5s6"
          >
            <div
              class="css-view-175oi2r"
              style="width: 0px;"
            >
              <button
                aria-disabled="true"
                aria-label="First Banner"
                class="css-view-175oi2r r-pointerEvents-12vffkv r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1wtj0ep r-maxWidth-q90dru r-overflow-1udh08x r-width-13qz1uu"
                data-testid="banner-tile"
                disabled=""
                role="button"
                style="background-color: rgb(248, 247, 252); border-top-left-radius: 20px; border-top-right-radius: 20px; border-bottom-right-radius: 20px; border-bottom-left-radius: 20px; opacity: 1;"
                tabindex="-1"
                type="button"
              >
                <div
                  class="css-view-175oi2r"
                  style="flex: 1;"
                >
                  <div
                    aria-label="First Banner"
                    class="css-view-175oi2r"
                  >
                    <div
                      class="css-text-146c3p1 r-fontSize-1ui5ee8 r-fontWeight-b88u0q r-marginBottom-6gpygo"
                      data-testid="banner-tile-title"
                      dir="auto"
                      style="color: rgb(0, 0, 0);"
                    >
                      First Banner
                    </div>
                  </div>
                  <div
                    aria-label="First Banner Description"
                    class="css-view-175oi2r"
                    data-testid="banner-tile-description"
                  >
                    <div
                      class="css-text-146c3p1 r-fontSize-1i10wst r-marginBottom-zd98yo"
                      dir="auto"
                      style="color: rgb(51, 51, 51);"
                    >
                      First Banner Description
                    </div>
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <button
        aria-label="Next slide"
        class="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73 r-alignItems-1awozwy r-borderRadius-10jnvle r-height-ur6pnr r-justifyContent-1777fci r-position-u8s1d r-top-1v2oles r-transform-1cjvy6m r-width-1mhb1uw r-zIndex-184en5c r-right-1nalo6"
        role="button"
        style="background-color: rgb(240, 240, 240);"
        tabindex="0"
        type="button"
      >
        <div
          class="css-view-175oi2r"
        >
          <svg
            class="lucide lucide-arrow-right"
            fill="none"
            height="20"
            stroke="#392ed7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="m12 5 7 7-7 7"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="css-view-175oi2r r-flexDirection-18u37iz r-justifyContent-1777fci r-marginTop-11c0sde"
    >
      <div
        class="css-view-175oi2r r-borderRadius-1xfd6ze r-height-3da1kt r-marginInline-1537yvj"
        style="background-color: rgb(57, 46, 215); width: 30px;"
      />
      <div
        class="css-view-175oi2r r-borderRadius-1xfd6ze r-height-3da1kt r-marginInline-1537yvj"
        style="background-color: rgb(192, 192, 192); width: 8px;"
      />
    </div>
  </div>
</div>
`;
